libfaad/tns.c

```diff
@@ -258,22 +258,20 @@ static void tns_ar_filter(real_t *spectrum, uint16_t size, int8_t inc, real_t *l
     /* state is stored as a double ringbuffer */
     real_t state[2*TNS_MAX_ORDER] = {0};
     int8_t state_index = 0;
+    real_t mul = 1;
 
 #ifdef FIXED_POINT
     if (exp >= 4)
         return;
+    mul = 1 << exp;
```

Comment:
Bit-shift on type `real_t`?

Reply:
In `FIXED_POINT` it is `int32_t`, and we use is as a plain factor (not `MUL_R`/`C`).
In other words we just moved left shift into a multiplicative constant.
But, perhaps it is better to make this variable implicitly integer. One sec.

Reply:
Fixed.